websrc/source/documents/community/CONTRIBUTING.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: "documents"
+page_title: "Contiv Community"
+sidebar_current: "community"
+description: |-
+  Contributing to Contiv
+---
+# Contributing to Contiv
+
+Submitting code changes is only one of the many ways to contribute.
+Reporting issues, proposing documentation and design changes,
+discussing use cases and proposing integration with other software
+from the ecosystem are more ways to contribute to Contiv.
+Contributors can also become maintainers or committers.
+
+All contributions are welcome, no matter how small or how big they are.
+
+Various repos may have more specific requirements for contribution - check out CONTRIBUTING.md
+files in respective repos.
+
+## Reporting Issues
+You can report an issue in a specific repo by 
+clicking `New Issue` in the respective repo.
+Issues should also be opened for feature requests.
+
+## Submitting pull requests for code or documentation changes
+Changes can be proposed by sending a pull request (PR). One of the committers
+will review the changes and provide feedback. The pull request will be merged
+into the master branch after discussion.
+
+Please make sure that the unit tests and the system tests pass before
+submitting the PR, as appropriate.
+Please keep in mind some changes might not be merged if committers
+decide they can't be merged.
+
+
+## Submitting proposals for major changes
+Please include `Proposal: ...` in the title of the issue if you wish to
+do significant refactoring of the code, to propose a new component or to
+introduce a major change. Marking the issue as a proposal will ensure that
+more people provide feedback as early as possible.
+
+Significant code changes submitted without an accompanying proposal might
+be rejected and not merged. Such significant submissions without a proposal are
+discouraged to avoid wasting time.
+
+
+After design discussions:
+- Fork the repository to your own public repository
+- Make the changes in your repository in a new branch
+- Add unit and system test cases for your code, as appropriate
+- Make sure existing tests and newly added tests pass, as appropriate
+- Rebase your branch on top of the latest master branch
+- Re-run the unit and system tests
+- Submit a pull request with the code changes
+- A discussion may take place on your pull request
+- Discussing while writing the code is also recommended
+- Requested changes should be made to the same branch on your fork
+- Changes should be force pushed to the same branch of the pull request
```

Comment:
Force pushing is not necessary unless you are rewriting history... the implication here is that you want them to keep their changes in a single commit.  Is that right, or is that something that is up to the maintainers of each repo and mentioned in CONTRIBUTING.md?

Reply:
I guess we need to add:
- rebase wrt master before resubmitting. This may require a force push to the same branch of the pull request.
@dseevr what do you think?

Reply:
@rhim Yeah, that sounds good.  If the goal is to make sure that PRs are on code which is as close to HEAD of master as possible, then calling that out makes sense.

I guess ensuring that PRs being merged are close to HEAD on master is also partially a committer responsibility.  We could actually add a CI gate for that, i.e., prevent merge if number of commits on master which aren't on the PR branch exceeds some cutoff.

Reply:
<img width="483" alt="screen shot 2017-09-21 at 15 40 13" src="https://user-images.githubusercontent.com/97581/30721941-46ca7bb2-9ee3-11e7-91c1-d8b8dddc45d4.png">

Looks like Github can enforce this for us.  :-)